### ISSUE TITLE
Impossible to emit CommonJS modules when targeting ES6

### ISSUE THREAD

**Issue by JannesMeyer** [TypeStrong/ts-loader#111]
It seems that the following line is [not necessary anymore with TypeScript 1.7](https://github.com/Microsoft/TypeScript/wiki/What%27s-new-in-TypeScript#support-for---target-es6-with---module):

https://github.com/TypeStrong/ts-loader/blob/ff15e76933e0f403e97f271f91577499add90f27/index.ts#L282

Removing it would allow us to target ES6 while still emitting CommonJS modules.

**Comment by jbrantly** [comment]
You're right, this needs to be changed.

**Comment by johnnyreilly** [comment]
Out of curiousity I implemented this change locally and found it broke 4 tests:

```
  1) es6resolveParent should have the correct output:

      Uncaught AssertionError: bundle.js is different between actual and expected
      + expected - actual

       /***/ function(module, exports) {

        "use strict";

      + Object.defineProperty(exports, "__esModule", {
      +   value: true
      + });
      +
        function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

        var BaseComponent = function BaseComponent() {
          _classCallCheck(this, BaseComponent);

      at E:\source\ts-loader\test\run.js:234:28
      at Array.forEach (native)
      at Watching.handler (E:\source\ts-loader\test\run.js:223:39)
      at Watching._done (E:\source\ts-loader\node_modules\webpack\lib\Compiler.js:81:7)
      at Watching.<anonymous> (E:\source\ts-loader\node_modules\webpack\lib\Compiler.js:61:18)
      at Compiler.emitRecords (E:\source\ts-loader\node_modules\webpack\lib\Compiler.js:282:37)
      at Watching.<anonymous> (E:\source\ts-loader\node_modules\webpack\lib\Compiler.js:58:19)
      at E:\source\ts-loader\node_modules\webpack\lib\Compiler.js:275:11
      at Compiler.applyPluginsAsync (E:\source\ts-loader\node_modules\webpack\node_modules\tapable\lib\Tapable.js:60:69)
      at Compiler.afterEmit (E:\source\ts-loader\node_modules\webpack\lib\Compiler.js:272:8)
      at Compiler.<anonymous> (E:\source\ts-loader\node_modules\webpack\lib\Compiler.js:267:14)
      at E:\source\ts-loader\node_modules\webpack\node_modules\async\lib\async.js:52:16
      at done (E:\source\ts-loader\node_modules\webpack\node_modules\async\lib\async.js:246:17)
      at E:\source\ts-loader\node_modules\webpack\node_modules\async\lib\async.js:44:16
      at E:\source\ts-loader\node_modules\webpack\node_modules\watchpack\node_modules\graceful-fs\graceful-fs.js:42:10
      at E:\source\ts-loader\node_modules\enhanced-resolve\node_modules\graceful-fs\graceful-fs.js:42:10
      at E:\source\ts-loader\node_modules\fs-extra\node_modules\graceful-fs\graceful-fs.js:42:10
      at E:\Source\ts-loader\node_modules\mocha\node_modules\glob\node_modules\graceful-fs\graceful-fs.js:104:5
      at FSReqWrap.oncomplete (fs.js:95:15)

  2) es6resolveParent should work with transpile:

      Uncaught AssertionError: bundle.js is different between actual and expected
      + expected - actual

       /***/ function(module, exports) {

        "use strict";

      + Object.defineProperty(exports, "__esModule", {
      +   value: true
      + });
      +
        function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

        var BaseComponent = function BaseComponent() {
          _classCallCheck(this, BaseComponent);

      at E:\source\ts-loader\test\run.js:234:28
      at Array.forEach (native)
      at Watching.handler (E:\source\ts-loader\test\run.js:223:39)
      at Watching._done (E:\source\ts-loader\node_modules\webpack\lib\Compiler.js:81:7)
      at Watching.<anonymous> (E:\source\ts-loader\node_modules\webpack\lib\Compiler.js:61:18)
      at Compiler.emitRecords (E:\source\ts-loader\node_modules\webpack\lib\Compiler.js:282:37)
      at Watching.<anonymous> (E:\source\ts-loader\node_modules\webpack\lib\Compiler.js:58:19)
      at E:\source\ts-loader\node_modules\webpack\lib\Compiler.js:275:11
      at Compiler.applyPluginsAsync (E:\source\ts-loader\node_modules\webpack\node_modules\tapable\lib\Tapable.js:60:69)
      at Compiler.afterEmit (E:\source\ts-loader\node_modules\webpack\lib\Compiler.js:272:8)
      at Compiler.<anonymous> (E:\source\ts-loader\node_modules\webpack\lib\Compiler.js:267:14)
      at E:\source\ts-loader\node_modules\webpack\node_modules\async\lib\async.js:52:16
      at done (E:\source\ts-loader\node_modules\webpack\node_modules\async\lib\async.js:246:17)
      at E:\source\ts-loader\node_modules\webpack\node_modules\async\lib\async.js:44:16
      at E:\source\ts-loader\node_modules\webpack\node_modules\watchpack\node_modules\graceful-fs\graceful-fs.js:42:10
      at E:\source\ts-loader\node_modules\enhanced-resolve\node_modules\graceful-fs\graceful-fs.js:42:10
      at E:\source\ts-loader\node_modules\fs-extra\node_modules\graceful-fs\graceful-fs.js:42:10
      at E:\Source\ts-loader\node_modules\mocha\node_modules\glob\node_modules\graceful-fs\graceful-fs.js:104:5
      at FSReqWrap.oncomplete (fs.js:95:15)

  3) issue92 should have the correct output:

      Uncaught AssertionError: bundle.js is different between actual and expected
      + expected - actual

       /***/ function(module, exports, __webpack_require__) {

        'use strict';

      - var submodule_1 = __webpack_require__(1);
      - exports['default'] = submodule_1['default'];
      + Object.defineProperty(exports, '__esModule', {
      +   value: true
      + });

      + function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }
      +
      + var _submoduleSubmodule = __webpack_require__(1);
      +
      + var _submoduleSubmodule2 = _interopRequireDefault(_submoduleSubmodule);
      +
      + exports['default'] = _submoduleSubmodule2['default'];
      + module.exports = exports['default'];
      +
       /***/ },
       /* 1 */
       /***/ function(module, exports) {

        "use strict";

      + Object.defineProperty(exports, "__esModule", {
      +   value: true
      + });
        exports["default"] = "Hello from submodule";
      + module.exports = exports["default"];

       /***/ }
       /******/ ]);

      at E:\source\ts-loader\test\run.js:234:28
      at Array.forEach (native)
      at Watching.handler (E:\source\ts-loader\test\run.js:223:39)
      at Watching._done (E:\source\ts-loader\node_modules\webpack\lib\Compiler.js:81:7)
      at Watching.<anonymous> (E:\source\ts-loader\node_modules\webpack\lib\Compiler.js:61:18)
      at Compiler.emitRecords (E:\source\ts-loader\node_modules\webpack\lib\Compiler.js:282:37)
      at Watching.<anonymous> (E:\source\ts-loader\node_modules\webpack\lib\Compiler.js:58:19)
      at E:\source\ts-loader\node_modules\webpack\lib\Compiler.js:275:11
      at Compiler.applyPluginsAsync (E:\source\ts-loader\node_modules\webpack\node_modules\tapable\lib\Tapable.js:60:69)
      at Compiler.afterEmit (E:\source\ts-loader\node_modules\webpack\lib\Compiler.js:272:8)
      at Compiler.<anonymous> (E:\source\ts-loader\node_modules\webpack\lib\Compiler.js:267:14)
      at E:\source\ts-loader\node_modules\webpack\node_modules\async\lib\async.js:52:16
      at done (E:\source\ts-loader\node_modules\webpack\node_modules\async\lib\async.js:246:17)
      at E:\source\ts-loader\node_modules\webpack\node_modules\async\lib\async.js:44:16
      at E:\source\ts-loader\node_modules\webpack\node_modules\watchpack\node_modules\graceful-fs\graceful-fs.js:42:10
      at E:\source\ts-loader\node_modules\enhanced-resolve\node_modules\graceful-fs\graceful-fs.js:42:10
      at E:\source\ts-loader\node_modules\fs-extra\node_modules\graceful-fs\graceful-fs.js:42:10
      at E:\Source\ts-loader\node_modules\mocha\node_modules\glob\node_modules\graceful-fs\graceful-fs.js:104:5
      at FSReqWrap.oncomplete (fs.js:95:15)

  4) issue92 should work with transpile:

      Uncaught AssertionError: bundle.js is different between actual and expected
      + expected - actual

       /***/ function(module, exports, __webpack_require__) {

        'use strict';

      - var submodule_1 = __webpack_require__(1);
      - exports['default'] = submodule_1['default'];
      + Object.defineProperty(exports, '__esModule', {
      +   value: true
      + });

      + function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }
      +
      + var _submoduleSubmodule = __webpack_require__(1);
      +
      + var _submoduleSubmodule2 = _interopRequireDefault(_submoduleSubmodule);
      +
      + exports['default'] = _submoduleSubmodule2['default'];
      + module.exports = exports['default'];
      +
       /***/ },
       /* 1 */
       /***/ function(module, exports) {

        "use strict";

      + Object.defineProperty(exports, "__esModule", {
      +   value: true
      + });
        exports["default"] = "Hello from submodule";
      + module.exports = exports["default"];

       /***/ }
       /******/ ]);

      at E:\source\ts-loader\test\run.js:234:28
      at Array.forEach (native)
      at Watching.handler (E:\source\ts-loader\test\run.js:223:39)
      at Watching._done (E:\source\ts-loader\node_modules\webpack\lib\Compiler.js:81:7)
      at Watching.<anonymous> (E:\source\ts-loader\node_modules\webpack\lib\Compiler.js:61:18)
      at Compiler.emitRecords (E:\source\ts-loader\node_modules\webpack\lib\Compiler.js:282:37)
      at Watching.<anonymous> (E:\source\ts-loader\node_modules\webpack\lib\Compiler.js:58:19)
      at E:\source\ts-loader\node_modules\webpack\lib\Compiler.js:275:11
      at Compiler.applyPluginsAsync (E:\source\ts-loader\node_modules\webpack\node_modules\tapable\lib\Tapable.js:60:69)
      at Compiler.afterEmit (E:\source\ts-loader\node_modules\webpack\lib\Compiler.js:272:8)
      at Compiler.<anonymous> (E:\source\ts-loader\node_modules\webpack\lib\Compiler.js:267:14)
      at E:\source\ts-loader\node_modules\webpack\node_modules\async\lib\async.js:52:16
      at done (E:\source\ts-loader\node_modules\webpack\node_modules\async\lib\async.js:246:17)
      at E:\source\ts-loader\node_modules\webpack\node_modules\async\lib\async.js:44:16
      at E:\source\ts-loader\node_modules\webpack\node_modules\watchpack\node_modules\graceful-fs\graceful-fs.js:42:10
      at E:\source\ts-loader\node_modules\enhanced-resolve\node_modules\graceful-fs\graceful-fs.js:42:10
      at E:\source\ts-loader\node_modules\fs-extra\node_modules\graceful-fs\graceful-fs.js:42:10
      at E:\Source\ts-loader\node_modules\mocha\node_modules\glob\node_modules\graceful-fs\graceful-fs.js:104:5
      at FSReqWrap.oncomplete (fs.js:95:15)
```

I'll be honest; my knowledge of what Webpack should be emitting is pretty non-existent.  Does this sound right?  I'm happy to amend the tests to comply with the changed emit but I'm not sure it is in this case.

**Comment by mpseidel** [comment]
+1 not sure how to fix the breaking tests either...

**Comment by jbrantly** [comment]
I'm currently working on the tests in #122 (but it's not quite ready yet)

**Comment by mpseidel** [comment]
great - thanks!

**Comment by jbrantly** [comment]
Apologies for the delay with the tests (been busy with newborn & other homelife stuff). I believe the tests are now good with #124.

**Comment by johnnyreilly** [comment]
Congratulations on the newborn!

**Comment by JannesMeyer** [comment]
Congratulations for your son/daughter!!!

**Comment by mpseidel** [comment]
@jbrantly congrats :)

**Comment by pawsong** [comment]
@jbrantly Thank you for your great job! I have seen that you wrote fix for this issue in [dont-force-module](https://github.com/TypeStrong/ts-loader/tree/dont-force-module) branch. Could you release a new version with this patch? As babel and TypeScript have different module interop mechanism, this will be very helpful when using babel as es6 -> es5 transpiler!

**Comment by jbrantly** [comment]
Apologies for the delay. This has been published in v0.8.0.
